examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/provisioning/DeviceProvisioningFragment.kt

```diff
@@ -135,36 +166,42 @@ class DeviceProvisioningFragment : Fragment() {
       if (wifi != null) {
         network = NetworkCredentials.forWiFi(NetworkCredentials.WiFiCredentials(wifi.ssid, wifi.password))
       }
+
       val thread = networkParcelable.threadCredentials
       if (thread != null) {
         network = NetworkCredentials.forThread(NetworkCredentials.ThreadCredentials(thread.operationalDataset))
       }
-      deviceController.setDeviceAttestationFailureCallback(DEVICE_ATTESTATION_FAILED_TIMEOUT
-      ) { devicePtr, errorCode ->
+
+      deviceController.setDeviceAttestationDelegate(DEVICE_ATTESTATION_FAILED_TIMEOUT,
+      { devicePtr, attestationInfo, errorCode ->
```

Comment:
Can this attestation delegate be pulled out into a class for reuse? It looks like this code is identical to the block above.